src/types/utility-types/refinement.ts

```diff
@@ -27,8 +27,11 @@ export class Refinement extends Type<any, any> {
         return inst
     }
 
-    is(value: any): value is any {
-        return this.type.is(value) && this.predicate(value)
+    validate(value: any, context: IContext): IValidationResult {
+        if (this.type.is(value) && this.predicate(value)) {
+            return []
```

Comment:
are these arrays modified after return? otherwise we could declare `EMPTY_ARRAY = Object.freeze([])` somewhere